src/arkode/arkode_io.c

```diff
@@ -1900,16 +2922,62 @@ int arkGetNumStepSolveFails(void* arkode_mem, long int* nncfails)
   }
   ark_mem = (ARKodeMem)arkode_mem;
 
+  /* Guard against use for time steppers that do not need an algebraic solver */
+  if (!ark_mem->step_supports_algebraic)
+  {
+    arkProcessError(ark_mem, ARK_STEPPER_UNSUPPORTED, __LINE__, __func__,
+                    __FILE__, "time-stepping module does not require an algebraic solver");
+    return (ARK_STEPPER_UNSUPPORTED);
+  }
+
   *nncfails = ark_mem->ncfn;
   return (ARK_SUCCESS);
 }
 
 /*---------------------------------------------------------------
-  arkGetUserData:
+  ARKodeGetNumLinSolvSetups:
+
+  Returns the current number of calls to the lsetup routine
+  ---------------------------------------------------------------*/
+int ARKodeGetNumLinSolvSetups(void* arkode_mem, long int* nlinsetups)
```

Comment:
I agree that this is essentially an identical question as for the nonlinear solvers, so let's see how the discussion evolves for that one, and this will follow suit.